### PR TITLE
Add static load method to Blob and Bucket, add tests, update example

### PR DESCRIPTION
**Pull request by mziccard**
This follows #253 and adds static `load` methods to both `Blob` and `Bucket`:
- `Blob.load(blobId)`
- `Blob.load(bucket, name)`
- `Bucket.load(bucket)`

All methods perform an RPC call and fetch latest information. Tests are added and storage example is updated accordingly.

### CODE REVIEW

File: gcloud-java-storage/src/main/java/com/google/gcloud/storage/Blob.java [hunk 1]
```diff
@@ -109,28 +109,27 @@ public Blob(Storage storage, BlobInfo info) {
...
+   * latest blob information.
    * 
    * @param storage the storage service used for issuing requests
    * @param blobId blob's identifier
    */
-  public Blob(Storage storage, BlobId blobId) {
-    this.storage = checkNotNull(storage);
-    this.info = BlobInfo.builder(blobId).build();
+  public static Blob load(Storage storage, BlobId blobId) {
+    BlobInfo info = storage.get(blobId);
```
Comment (aozarov) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/Blob.java]: Same comment for here and bucket:
(1)  javadoc should include "@throws StorageException upon failure"
(2) NPE if get returns `null`

Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/Blob.java]: Should we throw a special exception when `get` returns null? Or leave the NPE?

Comment (aozarov) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/Blob.java]: We could return `null` and document it (similar to `get`).

Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/Blob.java]: Fixed!

File: gcloud-java-examples/src/main/java/com/google/gcloud/examples/StorageExample.java [hunk 2]
```diff
@@ -121,7 +104,7 @@ public String params() {
       }
       Blob[] blobs = new Blob[args.length - 1];
       for (int i = 1; i < args.length; i++) {
-        blobs[i - 1] = new Blob(storage, args[0], args[i]);
+        blobs[i - 1] = Blob.load(storage, args[0], args[i]);
```
Comment (aozarov) [inline gcloud-java-examples/src/main/java/com/google/gcloud/examples/StorageExample.java]: Here and in other places in this example I don't think we handle well a `null` `Blob`.

Maybe the simplest way to deal with that is to make the `parse` methods return 
`BlobId` rather than `Blob`.
Which maybe makes sense as in some cases we don't need the `Blob`.

Comment (mziccard) [inline gcloud-java-examples/src/main/java/com/google/gcloud/examples/StorageExample.java]: I was planning to make major example updates in a different PR, also to use the static `get`, `update` and `delete` methods But I guess it's fine doing it here, I'm on it!

**Comment by mziccard** [comment]
I updated the storage example to use `BlobId` and `BlobInfo` (for `upload` and `signUrl`). I also did a minor change to readme and package-info examples (removing the use of `BlobInfo.of` and explicitly setting the `contentType`).

**Comment by aozarov** [comment]
Great, few more small comments.

**Comment by jfreebj** [comment]
OPT PUT PLEASE
Tiffany Burich

On Oct 15, 27 Heisei, at 9:42 AM, Marco Ziccardi <notifications@github.com> wrote:

I updated the storage example to use BlobId and BlobInfo (for upload and signUrl). I also did a minor change to readme and package-info examples (removing the use of BlobInfo.of and explicitly setting the contentType).

**Comment by aozarov** [comment]
Hey @jfreebj  what is the purpose of your comments?
